source/simulator/newton.cc

```diff
@@ -161,6 +161,239 @@ namespace aspect
+  unsigned int
+  NewtonHandler<dim>::
+  get_max_newton_line_search_iterations()
+  {
+    return max_newton_line_search_iterations;
+  }
+
+  template <int dim>
+  bool
+  NewtonHandler<dim>::
+  get_use_newton_residual_scaling_method()
+  {
+    return use_newton_residual_scaling_method;
+  }
+
+  template <int dim>
+  double
+  NewtonHandler<dim>::
+  get_maximum_linear_stokes_solver_tolerance()
+  {
+    return maximum_linear_stokes_solver_tolerance;
+  }
+
+  template <int dim>
+  std::string
+  NewtonHandler<dim>::
+  get_preconditioner_stabilization_string(PreconditionerStabilization preconditioner_stabilization)
+  {
+    switch (preconditioner_stabilization)
+      {
+        case PreconditionerStabilization::SPD:
+          return "SPD";
+          break;
+        case PreconditionerStabilization::PD:
+          return "PD";
+          break;
+        case PreconditionerStabilization::symmetric:
+          return "symmetric";
+          break;
+        case PreconditionerStabilization::none:
+          return "none";
+          break;
+      }
+  }
+
+  template <int dim>
+  std::string
+  NewtonHandler<dim>::
+  get_velocity_block_stabilization_string(VelocityBlockStabilization velocity_block_stabilization)
+  {
+    switch (velocity_block_stabilization)
+      {
+        case VelocityBlockStabilization::SPD:
+          return "SPD";
+          break;
+        case VelocityBlockStabilization::PD:
+          return "PD";
+          break;
+        case VelocityBlockStabilization::symmetric:
+          return "symmetric";
+          break;
+        case VelocityBlockStabilization::none:
+          return "none";
+          break;
+      }
+  }
+
+  template <int dim>
+  void
+  NewtonHandler<dim>::
+  declare_parameters (ParameterHandler &prm)
+  {
+    prm.enter_subsection ("Solver parameters");
+    {
+      prm.enter_subsection ("Newton solver parameters");
+      {
+        prm.declare_entry ("Nonlinear Newton solver switch tolerance", "1e-5",
+                           Patterns::Double(0,1),
+                           "A relative tolerance with respect to the residual of the first "
+                           "iteration, up to which the nonlinear Picard solver will iterate, "
+                           "before changing to the newton solver.");
+
+        prm.declare_entry ("Max pre-Newton nonlinear iterations", "10",
+                           Patterns::Integer (0),
+                           "The maximum number of Picard nonlinear iterations to be performed "
+                           "before switching to Newton iterations.");
+
+        prm.declare_entry ("Max Newton line search iterations", "5",
+                           Patterns::Integer (0),
+                           "The maximum number of line search iterations allowed. If the "
+                           "criterion is not reached after this iteration, we apply the scaled "
+                           "increment to the solution and continue.");
```

Comment:
-> apply the scaled increment even though it does not satisfy the necessary criteria and simply continue with the next Newton iteration